### PR TITLE
FIX: Add endianness missing flag when reading data

### LINKED ISSUE #8688: 0.15.0 test_stata  strikes back on various less common platforms on
e.g. 64bit big-endian
https://buildd.debian.org/status/fetch.php?pkg=pandas&arch=s390x&ver=0.15.0-1&stamp=1414699796
and others
https://buildd.debian.org/status/package.php?p=pandas
(some have other failures as well, yet to find a moment to "analyze", but e.g. hurd-i386 fails test_precise_conversion)

could/should all stata tests be skipped on big-endians?

### PR DESCRIPTION
**Pull request by bashtage**
Added endianess flat to data type to allow data to be read cross platforms

closes #8688

### CODE REVIEW

File: pandas/io/stata.py [hunk 1]
```diff
@@ -1189,13 +1190,18 @@ def data(self, convert_dates=True, convert_categoricals=True, index=None,
...
                 dtype.append(('s' + str(i), 'S' + str(typ)))
         dtype = np.dtype(dtype)
         read_len = count * dtype.itemsize
         self.path_or_buf.seek(self.data_location)
         data = np.frombuffer(self.path_or_buf.read(read_len),dtype=dtype,count=count)
+        # if necessary, swap the byte order to native here
+        if self.byteorder != self._native_byteorder:
+            print(data.dtype)
+            data = data.byteswap().newbyteorder()
+            print(data.dtype)
```
Comment (yarikoptic) [inline pandas/io/stata.py]: prints left from debugging or really intended for some reason?

Comment (bashtage) [inline pandas/io/stata.py]: Should not be there...

**Comment by bashtage** [comment]
@yarikoptic @jreback I think this one is ready.

**Comment by jreback** [comment]
@yarikoptic ok on these platforms?

**Comment by jreback** [comment]
@bashtage IIRC the tests are disabled on non-small so is this actually tested?

**Comment by bashtage** [comment]
All of the tests were re-enabled for big-endian a few releases ago.

There are only three tests that are skipped on any platform:

https://github.com/pydata/pandas/blob/master/pandas/io/tests/test_stata.py#L256

I would like to kill these in a different PR since they rely on proprietary data and so will never be re-enabled.

**Comment by jreback** [comment]
ok, ping on green

**Comment by bashtage** [comment]
@jreback It is ready.

**Comment by jreback** [comment]
thanks!

**Pull request merged**